### ISSUE TITLE
Trucks are dumping in locations not assigned

### ISSUE THREAD

**Issue by utdrmac** [MaFi-Games/Captain-of-Industry-issues#1165]
### Issue description

There are two mining towers with adjacent mining locations. Both locations have mining and dumping zones. See screenshot tower_areas.png  A truck from right tower has dirt, rocks, and iron. It has unloaded the iron into the assigned tower's storage. But for some wrong reason, it is now driving over to another mine's area to dump the dirt/rocks because there are no available areas in his mining area.

This is the incorrect behavior per descriptions of tower assignment. This truck is assigned to right tower. If it cannot dump, it should not go looking for another tower's area; it should give warning that it can't dump and remain close to assigned tower.

Globally, all dump items have been removed.

### Reproduction steps

Create two mining towers and assign a large area. Mark mining and dumping locations in each area. If dumping locations are unavailable, trucks will go searching for the closest location.

### Game version and build number

Update 1 v0.5.2a

### Operating system

Windows

### Attachments

![pickup_destination](https://github.com/MaFi-Games/Captain-of-Industry-issues/assets/4413670/e8f6090d-5110-4f47-8abf-ca128bd9f650)
![tower_areas](https://github.com/MaFi-Games/Captain-of-Industry-issues/assets/4413670/5825a833-55fb-4549-b7ad-b63e3f8b2c44)
[Captains Log.zip](https://github.com/MaFi-Games/Captain-of-Industry-issues/files/11854558/Captains.Log.zip)

**Comment by utdrmac** [comment]
Captains Log.zip is saved game file at location of screenshots

**Comment by MaFi-Games** [comment]
I don't think this is incorrect behavior. The fact the truck dumps into another mine tower area is because that target tower does not have any import routes assigned, so all its imports (including dumping on its designations) are unrestricted. To make this work you would need a toggle on that Mine tower area to restrict its dumping zones to its mining trucks only when no import routes are assigned.

**Comment by MaFi-Games** [comment]
I'm not going to argue if it's logical or not. But there are years of experience with the system, and enforcing tower only dumping designations brings an entire new set of issues. Such as that having a single fulfilled dumping designation in a tower suddenly changes dumping pattern of all its assigned trucks. Or that it become less intuitive for new players that don't see this behavior explicitely documented anywhere.

> I don't see this option anywhere in the MCT window to restrict dumping zones. (Update 1 | v0.5.3c)
It's not in the game. I was just trying to explain what would neccessary to make it work.

**Comment by JamieRCHI** [comment]
Maybe close this issue as 'By Design'.
